### PR TITLE
Captian we have to cook: Rework of the morenarcotics module

### PR DESCRIPTION
**Pull request by Paxilmaniac**
## About The Pull Request

What am I doing:

So you know all those drugs we added, cocaine, heroin, thc, you know that shit.

That? That's boring, that's lame, that's also a lot of stuff of questionable quality and dubious balance. We're past the year 2500 surely life ruining addictive substances have advanced in technology by now.

So what do we do about it? We remove all of the stuff we added to make place for something a bit more fitting for a setting that's a few hundred years in the future.

Behold, John Tide's cookbook of space age substances:

1. Crystalline firefighting foam
Looking for a high? No skills in growing plants to smoke? Don't worry buddy we've got you covered. Just get yourself some firefighting foam in a container and heat that shit up. The crystals this makes can be rolled up into joints for your smoking pleasure. If you're a madman you could also just crunch on the crystals themselves, but I wouldn't suggest that.

image

2. Odious Puffpowder
Odious puffballs, yeah you know those invasive purple mushrooms that spit spore toxin everywhere. Normally nothing but that, an invasive pest, but some enterprising individuals have figured that you can mix the spore toxin they produce with a bit of lye to make a fine powder. Now, this powder alone? Stuff's near useless, I mean if you're desperate it has some extremely minor medicinal properties, but there's a better use for that stuff. Find yourself an oven and bake the powder, if you give it some time you'll have yourself a fresh new block of SMARTs.

The hell is a SMARTs you're asking? These big blocks of refined purple stuff can be cut up and eaten to give yourself a good boost on how quickly you pick up new skills. Stuff's gross as hell though, I'd bring some hot sauce.

image

3. Demoneye
Demoneye's where we're getting into the big leagues. Real popular with the big crime types on mars especially, you spray the stuff in your eyes and it'll give you some crazy strength for a short time. Downsides? Ah don't worry about 'em, if you're taking this stuff you probably don't care about what happens after.

How do you make it? Pretty simple, a mix of blood, liquid plasma, and some ephedrine should work, should.

image

4. TWitch
TWitch is the top of the line preemo shit everyone's after. And by everyone, I mean people like you. Inject this stuff and you'll be on a whole different level of perception and ability, at least that's what the guy who taught me the recipe said. I've heard some good rumors though that taking this stuff lets people move faster, hit quicker, and usually gives a nasty nosebleed after.

The recipe for the stuff is the most out there I've seen for street drugs yet, you've gotta mix together some bluespace dust, enriched liquid electricity, and impedrezene, whatever that last one is.

image

## How This Contributes To The Skyrat Roleplay Experience

Just about all of the drugs we added in this module felt either like they were completely uninteresting and were just dude weed no way, or were tools of questionable balance (PCP ITS PCP FUCK YOU).

Rather than just do my original plan of removing the module, I thought I'd at least try improve don't remove by giving us a new selection of substances with a more sci-fi theme, and with more interesting effects than "Makes you immortal for 30 seconds but shakes your screen :("

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
image
image
image
image
image
image

</details>

## Changelog

:cl:
add: In place of the removed skyrat drugs, a smaller selection of more sci-fi themed narcotics have been added, with varying amounts of benefits and much worse downsides.
del: Most of the skyrat added drugs and related changes have been removed and undone, no more cocaine, no more heroin, so on.
/:cl:

### CODE REVIEW

**Comment by RatFromTheJungle** [comment]
The demon-eye adding +5 punch damage is also something that really-really shouldn't happen, because that's like baton levels of stamina per punch.


the cannabis changes **especially** are just a flat-downgrade to an inferior version, we went away from the TG changes for a reason.

**Comment by Stalkeros2** [comment]
GOTTA GET A GRIP

**Comment by Paxilmaniac** [comment]
> for fairly-lethal side-effects, which if we want casual drug-use, is about the opposite of what they'd need.

10 stamina loss every now and then if you overdose on extinguisher crystals
image

Spore toxin does a whopping zero damage and blurs your eyes
Medicine failure heals you for 0.5 of a random damage whenever it ticks
Heparin does nothing unless you are actively bleeding
image

Gee these sure are lethal side effects.

**Comment by RatFromTheJungle** [comment]
> > for fairly-lethal side-effects, which if we want casual drug-use, is about the opposite of what they'd need.
> 
> 10 stamina loss every now and then if you overdose on extinguisher crystals image


image
 and histamines, don't forget those, which from what I can see skimming this, with *2 chems-to-histamines, hitting that 30u od wouldn't be particularly hard

**Comment by Paxilmaniac** [comment]
> > > for fairly-lethal side-effects, which if we want casual drug-use, is about the opposite of what they'd need.
> > 
> > 
> > 10 stamina loss every now and then if you overdose on extinguisher crystals image
> 
> image and histamines, don't forget those, which from what I can see skimming this, with *2 chems-to-histamines, hitting that 30u od wouldn't be particularly hard

You only get hitamines if you take it some way that isn't either smoking it or just eating it.

**Comment by RatFromTheJungle** [comment]
> Spore toxin does a whopping zero damage and blurs your eyes Medicine failure heals you for 0.5 of a random damage whenever it ticks Heparin does nothing unless you are actively bleeding image
> 
> Gee these sure are lethal side effects.

I trimmed off the second one because I mentally processed it as the mold toxin

**Comment by RatFromTheJungle** [comment]
> > > 10 stamina loss every now and then if you overdose on extinguisher crystals image
> > 
> > 
> > image and histamines, don't forget those, which from what I can see skimming this, with *2 chems-to-histamines, hitting that 30u od wouldn't be particularly hard
> 
> You only get hitamines if you take it some way that isn't either smoking it or just eating it.

Is smoking stuff still on ingest? 

it is, cursed, already, disregard that one

i'll trim the comment down again (v8)

File: modular_skyrat/master_files/code/modules/uplink/uplink_items.dm [hunk 1]
```diff
@@ -200,15 +200,6 @@
 	item = /obj/item/syndie_glue
 	cost = 3
 
-/datum/uplink_item/device_tools/syndikush
```
Comment (Zonespace27) [inline modular_skyrat/master_files/code/modules/uplink/uplink_items.dm]: 😔 gone too soon

**Comment by Zenitheevee** [comment]
... so like genuine question-
Do we _need_ to remove all the old drugs?
Like i just kinda added a lot of them to the smuggler sat- 
Edit: Ope. Just did a double check on files.

File: modular_skyrat/modules/morenarcotics/code/twitch.dm [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+/datum/reagent/drug/twitch/proc/on_movement(mob/living/carbon/our_guy)
+	SIGNAL_HANDLER
+	new /obj/effect/temp_visual/decoy/twitch_afterimage(our_guy.loc, our_guy)
+
+/datum/reagent/drug/twitch/on_mob_life(mob/living/carbon/our_guy, delta_time, times_fired)
+	. = ..()
+
+	constant_dose_time += delta_time
+
+	for(var/possible_kronk in our_guy.reagents.reagent_list) // Kronkaine, another heart-straining drug, could cause problems if mixed with this
```
Comment (Zonespace27) [inline modular_skyrat/modules/morenarcotics/code/twitch.dm]: couldn't 121-122 be `if(locate(/datum/reagent/drug/kronkaine) in our_guy.reagents.reagent_list)`?

**Comment by Paxilmaniac** [comment]
The mapping tag is a bug report tag not a pr tag stop putting it on everything.

File: modular_skyrat/modules/morenarcotics/code/twitch.dm (hunk 2 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/morenarcotics/code/twitch.dm]: Probably could be

**Comment by theselfish** [comment]
The car replaced the horse and carriage, yet they still exist.
Most armies since WW2 phased out cavalry divisions, yet countries like India, China, Argentina and Chile still have them.

I very much doubt drugs that are still a big issue today, and will be for many, many years will suddenly vanish because it's the future.

Heroin has been used since the days of Mesopotamia. The modern synthesis of a lot of drugs (heroin and meth from the top of my head) were only in the 1800s, the former being 1874 and the later being 1887. That's only 140ish years, recent in the grand scheme of things.

**Comment by Paxilmaniac** [comment]
Weird way to say you don't like my changes but ok.

**Comment by Stonetear** [comment]
> Weird way to say you don't like my changes but ok.

I like the idea of both new and old drugs!  I'd rather not see the opium and cocaine removed.  I personally dislike synthetic drugs, and none of my characters would take them, but that's just me.

Perhaps some of the drug vials should fit in a boot like pump-up injectors.  The problem is that with stun combat, anything that doesn't give either stamina regen or stun baton resistance relegates it to roleplaying use.  (which isn't necessarily bad, just something to consider)

**Comment by PCPort** [comment]
> Perhaps some of the drug vials should fit in a boot like pump-up injectors. The problem is that with stun combat, anything that doesn't give either stamina regen or stun baton resistance relegates it to roleplaying use. (which isn't necessarily bad, just something to consider)

I mean... You could also just not get hit, and reap the benefits of having superhuman speed.

**Comment by Zenitheevee** [comment]
Right so
I still genuinely think removing the old stuff is a bit... :/ 
I do _Like_ the new stuff being listed, But getting past that, There are some... Fair concerns i did kinda use to justify the smuggler satchel change recently.

The main concern being: What exactly are these new drugs _worth_? 
I will genuinely say i dont like the removal of old drugs because as a _smugglable comodity_ they were great money producers. However in bringing that point up, i see no export changes with either Drug item sets, which i find odd.

Would there be any way to list *roughly* how much these proposed drugs would be worth on the cargo shuttle?

edit: also, small side note... Why do we have a cannabis drug.. that only gets used with the ultimate sub-item?

**Comment by Paxilmaniac** [comment]
> Right so I still genuinely think removing the old stuff is a bit... :/ I do _Like_ the new stuff being listed, But getting past that, There are some... Fair concerns i did kinda use to justify the smuggler satchel change recently.
> 
> The main concern being: What exactly are these new drugs _worth_? I will genuinely say i dont like the removal of old drugs because as a _smugglable comodity_ they were great money producers. However in bringing that point up, i see no export changes with either Drug item sets, which i find odd.
> 
> Would there be any way to list _roughly_ how much these proposed drugs would be worth on the cargo shuttle?
> 
> edit: also, small side note... Why do we have a cannabis drug.. that only gets used with the ultimate sub-item?

The only ones that would have value would be the two that don't have special injectors, and only because they are subtypes of food items so they'd prooobably have poor export value that way. If you want to make money selling drugs then you do so by selling to the other people playing the game, not to the cargo shuttle.

**Comment by theselfish** [comment]
> Weird way to say you don't like my changes but ok.

The additions are fine.
Removing the old drugs is what I dislike.
If their balance is an issue, change that. If their effects are annoying, change that.
Removing them because 'they're not sci-fi' is honestly bizarre to me.

**Review by tf-4 — CHANGES_REQUESTED** [review]
I'm not sure why we would need to remove the old stuff. It needs to be changed, yes, but not really removed.

**Comment by Paxilmaniac** [comment]
Fuck it I'll come back to this with the rebalance later

**Pull request closed without merging**